lldb/source/Interpreter/CommandInterpreter.cpp

```diff
@@ -2076,7 +2077,11 @@ bool CommandInterpreter::HandleCommand(const char *command_line,
     }
 
     ElapsedTime elapsed(execute_time);
-    cmd_obj->Execute(remainder.c_str(), result);
+    size_t nchar = real_original_command_string.find(remainder);
+    std::optional<uint16_t> pos_in_cmd;
+    if (nchar != std::string::npos)
+      pos_in_cmd = nchar + GetDebugger().GetPrompt().size();
+    cmd_obj->Execute(remainder.c_str(), pos_in_cmd, result);
```

Comment:
Instead of updating every `DoExecute` implementations, it would be reasonable to have make this an optional member of the `CommandObject` base class so it can be accessed by the derived `CommandObject` classes that gonna use it.


The optional could be set in the non-virtual `CommandObject::Execute` command, since that's the one that calls the derived class `DoExecute` or may be by adding a setter to the `CommandObject` base class. 

This would greatly reduce the size of this patch.

Reply:
I think expr is probably the only command that is likely to ever echo its command string into the output.  That's not something most commands would do.  So this doesn't seem like a feature that should have as much prominence as changing the API gives it.  I don't want to have to worry about what this means when I'm writing a new command, given the answer is almost surely to be "not relevant".

Reply:
Also, how does this work if I run

expr -i 0 -u 0 -- not a valid expression

The command string here is `-i 0 -u 0 -- not a valid expression` but you probably don't ever echo that full line onto the diagnostic, so either you are fudging and only matching the args part of the command, or the docs aren't right here.

Reply:
This seems a very expr specific function and not a generally useful for commands function.

Reply:
> I think expr is probably the only command that is likely to ever echo its command string into the output. That's not something most commands would do. So this doesn't seem like a feature that should have as much prominence as changing the API gives it. I don't want to have to worry about what this means when I'm writing a new command, given the answer is almost surely to be "not relevant".

I don't have a strong option about this (as long as we don't update all the `DoExecute` implementations), but I was under the impression that in the future, some commands could make use of it to point directly to what caused an error in their invocation. But again, we can have it be part of the expression for now.

Reply:
@jimingham @medismailben I hopefully have resolved both of your concerns by storing the original command string as a member in CommandObject. This way the `expr -i 0 -u 0 -- not a valid expression` works.
```
(lldb) expr -i 0 -u 0 -- not a valid expression
                         ^
                         ╰─ error: use of undeclared identifier 'not'
```